ir_Lego_PF.cpp

```diff
@@ -44,3 +43,55 @@ void IRsend::sendLegoPowerFunctions(uint16_t data, bool repeat)
 }
 
 #endif // SEND_LEGO_PF
+
+//+=============================================================================
+#if DECODE_LEGO_PF
+bool  IRrecv::decodeLegoPowerFunctions (decode_results *results) {
+	static LegoPfBitStreamEncoder bitStreamEncoder;
+	uint16_t data   = 0;
+	int  offset = 1;
+
+	// Check SIZE
+	if (irparams.rawlen < 2 * (bitStreamEncoder.MESSAGE_BITS) - 1)  return false ;  // exclude STOP pause
+
+	// Check START_BIT Mark/Space
+	if (!MATCH_MARK (results->rawbuf[offset++], bitStreamEncoder.IR_MARK_DURATION - 58)) return false ;

```

Comment:
Thanks much for the insight. There are plans to re-work IRRemote. Let's keep your current implementation and adapt it when the restructured library is available.